components/dashboards-web-component/src/viewer/components/DashboardRenderer.jsx

```diff
@@ -85,6 +89,23 @@ export default class DashboardRenderer extends Component {
             this.triggerThemeChangeEvent(nextProps.theme);
             return true;
         }
+        if (this.props.theme.name === 'light') {
```

Comment:
Doesn't this get covered by the `if` condition in above line 87?

Reply:
How about changing condition in line 87 to `this.props.theme.name !== nextProps.theme.name`

Reply:
It doesn't seem working.

Reply:
`this.props.theme.name === 'light'` feels like a hack. If the theme changed from light to dark, then this condition will be false and `this.triggerThemeChangeEvent(nextProps.theme);` in the following line 96 won't get executed. We need to trigger the theme change event for every time theme gets chaged (from dark to light as well as from light to dark)

Reply:
Agree

Reply:
Lets discuss this further, and we can fix this later. @kts-desilva Please add a TODO for this.

Reply:
Ok